build-system/corekit/publishing.gradle

```diff
@@ -12,114 +13,174 @@ task sourceJar(type: Jar) {
-                    dependencyNode.appendNode('version', it.version)
+            pom {
+                name = libraryNameCoreKit
+                description = libraryDescription
+                url = gitUrl
+                licenses {
+                    license {
+                        name = licenseName
+                        url = licenseUrl
+                    }
                 }
+                developers {
+                    developer {
+                        id = developerId
+                        name = developerName
+                        email = developerEmail
+                    }
+                }
+                scm {
+                    connection = scmConnection
+                    developerConnection = scmDeveloperConnection
+                    url = scmUrl
+                }
+                withXml {
+                    def dependenciesNode = asNode().appendNode('dependencies')
 
-                configurations.productionImplementation.allDependencies.each {
-                    def dependencyNode = dependenciesNode.appendNode('dependency')
-                    dependencyNode.appendNode('groupId', it.group)
-                    dependencyNode.appendNode('artifactId', it.name)
-                    dependencyNode.appendNode('version', it.version)
+                    configurations.implementation.allDependencies.each {
+                        def dependencyNode = dependenciesNode.appendNode('dependency')
+                        dependencyNode.appendNode('groupId', it.group)
+                        dependencyNode.appendNode('artifactId', it.name)
+                        dependencyNode.appendNode('version', it.version)
+                    }
+
+                    configurations.productionImplementation.allDependencies.each {
+                        def dependencyNode = dependenciesNode.appendNode('dependency')
+                        dependencyNode.appendNode('groupId', it.group)
+                        dependencyNode.appendNode('artifactId', it.name)
+                        dependencyNode.appendNode('version', it.version)
+                    }
                 }
             }
         }
     }
-}
 
-bintray {
-    user = System.getenv('BINTRAY_USER')
-    key = System.getenv('BINTRAY_KEY')
-    configurations = ['archives']
-    pkg {
-        repo = bintrayRepo
-        name = project.name
-        userOrg = bintrayOrg
-        licenses = allLicenses
-        vcsUrl = gitUrl
-        desc = libraryDescription
-        websiteUrl = siteUrl
-        publish = true
-        publicDownloadNumbers = true
+    repositories {
+        maven {
+            name = mavenRepo
+            url = mavenUrl
+            credentials {
+                username ossrhUsername
+                password ossrhPassword
+            }
+        }
     }
-    publications = ['BintrayProduction', 'BintraySandbox', 'BintrayUat']
-}
\ No newline at end of file
+}
+
+signing {
+    sign publishing.publications
+}
```

Comment:
Sorry I am not sure about what the signing do, is this required for uploading to maven? will other dev able to use the signed sdk library?

Reply:
same comment above